### ISSUE TITLE
Crash in 'flutter create' (?)

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#52702]
@christopherfujino This is on `dev/1.15.21` with a command `create --ios-language swift --android-language kotlin .`. Maybe there is a bad interaction when the artifacts are incomplete in `FlutterCommand.verifyAndRunCommand()` ?

```
ArgumentError: Invalid argument(s): Cannot find executable for E:\[...]\bin\cache\artifacts\libimobiledevice\idevicescreenshot
  | at _getExecutable | (local_process_manager.dart:127 )
  | at LocalProcessManager.runSync | (local_process_manager.dart:94 )
  | at _DefaultProcessUtils.exitsHappySync | (process.dart:541 )
  | at IMobileDevice.isInstalled | (mac.dart:39 )
  | at IOSWorkflow.canListDevices | (ios_workflow.dart:20 )
  | at IOSDevices.canListAnything | (devices.dart:38 )
  | at DeviceManager.canListAnything.<anonymous closure> | (device.dart:150 )
  | at Iterable.any | (iterable.dart:388 )
  | at DeviceManager.canListAnything | (device.dart:150 )
  | at _DefaultDoctorValidatorsProvider.validators | (doctor.dart:99 )
  | at Doctor.validators | (doctor.dart:158 )
  | at Doctor.startValidatorTasks | (doctor.dart:164 )
  | at Doctor.diagnose | (doctor.dart:270 )
  | at CreateCommand.runCommand | (create.dart:493 )
  | at _rootRunUnary | (zone.dart:1134 )
  | at _CustomZone.runUnary | (zone.dart:1031 )
  | at _FutureListener.handleValue | (future_impl.dart:140 )
  | at Future._propagateToListeners.handleValueCallback | (future_impl.dart:682 )
  | at Future._propagateToListeners | (future_impl.dart:711 )
  | at Future._completeWithValue | (future_impl.dart:526 )
  | at _AsyncAwaitCompleter.complete | (async_patch.dart:34 )
  | at _completeOnAsyncReturn | (async_patch.dart:293 )
  | at CreateCommand._generateApp | (create.dart )
  | at _rootRunUnary | (zone.dart:1134 )
  | at _CustomZone.runUnary | (zone.dart:1031 )
  | at _FutureListener.handleValue | (future_impl.dart:140 )
  | at Future._propagateToListeners.handleValueCallback | (future_impl.dart:682 )
  | at Future._propagateToListeners | (future_impl.dart:711 )
  | at Future._completeWithValue | (future_impl.dart:526 )
  | at _AsyncAwaitCompleter.complete | (async_patch.dart:34 )
  | at _completeOnAsyncReturn | (async_patch.dart:293 )
  | at FlutterProject.ensureReadyForPlatformSpecificTooling | (project.dart )
  | at _rootRunUnary | (zone.dart:1134 )
  | at _CustomZone.runUnary | (zone.dart:1031 )
  | at _FutureListener.handleValue | (future_impl.dart:140 )
  | at Future._propagateToListeners.handleValueCallback | (future_impl.dart:682 )
  | at Future._propagateToListeners | (future_impl.dart:711 )
  | at Future._completeWithValue | (future_impl.dart:526 )
  | at _AsyncAwaitCompleter.complete | (async_patch.dart:34 )
  | at _completeOnAsyncReturn | (async_patch.dart:293 )
  | at injectPlugins | (plugins.dart )
  | at _rootRunUnary | (zone.dart:1134 )
  | at _CustomZone.runUnary | (zone.dart:1031 )
  | at _FutureListener.handleValue | (future_impl.dart:140 )
  | at Future._propagateToListeners.handleValueCallback | (future_impl.dart:682 )
  | at Future._propagateToListeners | (future_impl.dart:711 )
  | at Future._completeWithValue | (future_impl.dart:526 )
  | at Future._asyncComplete.<anonymous closure> | (future_impl.dart:556 )
  | at _rootRun | (zone.dart:1126 )
  | at _CustomZone.run | (zone.dart:1023 )
  | at _CustomZone.runGuarded | (zone.dart:925 )
  | at _CustomZone.bindCallbackGuarded.<anonymous closure> | (zone.dart:965 )
  | at _microtaskLoop | (schedule_microtask.dart:43 )
  | at _startMicrotaskLoop | (schedule_microtask.dart:52 )
  | at _runPendingImmediateCallback | (isolate_patch.dart:118 )
  | at _RawReceivePortImpl._handleMessage | (isolate_patch.dart:169 )
```

**Comment by christopherfujino** [comment]
I'll take a look

**Comment by jonahwilliams** [comment]
It looks like `flutter create` might be running some doctor checks after the fact, but the create command doesn't declare the same required artifacts as doctor. The fix would be to point one set at the other.

**Comment by christopherfujino** [comment]
`libimobiledevice` is universal though, which should always be updated in `verifyThenRunCommand()` https://github.com/flutter/flutter/blob/v1.15.21/packages/flutter_tools/lib/src/runner/flutter_command.dart#L691 which should run before `| at CreateCommand.runCommand | (create.dart:493 )`

**Comment by jonahwilliams** [comment]
ahh, good point. I saw a similar issue with the android artifacts and got confused

**Comment by dnfield** [comment]
Why is it looking for iOS artifacts on a Windows machine? :\

**Comment by dnfield** [comment]
> E:\[...]\bin\cache\artifacts\libimobiledevice\idevicescreenshot

**Comment by christopherfujino** [comment]
Oh shoot!

**Comment by christopherfujino** [comment]
So this change https://github.com/flutter/flutter/commit/38a8845481b320b2fd4c52b2b55e2e0360e88a62 landed on dev 1.15.21, but I don't see how that would lead to a regression...they both should resolve to `globals.platform.isMacOS`

**Comment by christopherfujino** [comment]
Not sure how we got from `IOSWorkflow.canListDevices (ios_workflow.dart:20)` to `IMobileDevice.isInstalled (mac.dart:39)`, but I think the real problem here is that `ArgumentError` is not an `Exception`, and we're only catching `Exception` in `exitsHappySync`, which I will make a PR for.
